packages/protocol-http/src/httpHandlerInitialization.ts

```diff
@@ -0,0 +1,35 @@
+import type { Agent as hAgent } from "http";
+import type { Agent as hsAgent } from "https";
+
+/**
+ *
+ * This type represents an alternate client constructor option for the entry
+ * "requestHandler". Instead of providing an instance of a requestHandler, the user
+ * may provide the requestHandler's constructor options for either the
+ * NodeHttpHandler or FetchHttpHandler.
+ *
+ * For other RequestHandlers, constructor parameter passthrough is not available.
+ *
+ * @public
+ */
+export type RequestHandlerParams = NodeHttpHandlerOptions | FetchHttpHandlerOptions;
```

Comment:
should these types and interfaces live in `@smithy/types`, and the `@smithy/*-http-handler` packages import them so there is no duplication?

Reply:
yes, let's do that